python/semantic_kernel/connectors/ai/onnx/services/onnx_gen_ai_chat_completion.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+import sys
+from collections.abc import AsyncGenerator
+from typing import Any, ClassVar
+
+if sys.version_info >= (3, 12):
+    from typing import override  # pragma: no cover
+else:
+    from typing_extensions import override  # pragma: no cover
+
+
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.ai.chat_completion_client_base import ChatCompletionClientBase
+from semantic_kernel.connectors.ai.onnx.onnx_gen_ai_prompt_execution_settings import OnnxGenAIPromptExecutionSettings
+from semantic_kernel.connectors.ai.onnx.onnx_gen_ai_settings import OnnxGenAISettings
+from semantic_kernel.connectors.ai.onnx.services.onnx_gen_ai_completion_base import OnnxGenAICompletionBase
+from semantic_kernel.connectors.ai.onnx.utils import ONNXTemplate, apply_template
+from semantic_kernel.connectors.ai.prompt_execution_settings import PromptExecutionSettings
+from semantic_kernel.contents import (
+    ChatHistory,
+    ChatMessageContent,
+    ImageContent,
+    StreamingChatMessageContent,
+    TextContent,
+)
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.exceptions import ServiceInitializationError, ServiceInvalidExecutionSettingsError
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+@experimental_class
+class OnnxGenAIChatCompletion(ChatCompletionClientBase, OnnxGenAICompletionBase):
+    """OnnxGenAI text completion service."""
+
+    template: ONNXTemplate
+    SUPPORTS_FUNCTION_CALLING: ClassVar[bool] = False
+
+    def __init__(
+        self,
+        template: ONNXTemplate,
+        ai_model_path: str | None = None,
+        ai_model_id: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+    ) -> None:
+        """Initializes a new instance of the OnnxGenAITextCompletion class.
+
+        Args:
+            template (ONNXTemplate): The chat template configuration.
+            ai_model_path (str): Local path to the ONNX model Folder.
+            ai_model_id (str, optional): The ID of the AI model. Defaults to None.
+            env_file_path (str | None): Use the environment settings file as a fallback
+                to environment variables.
+            env_file_encoding (str | None): The encoding of the environment settings file.
+        """
+        try:
+            settings = OnnxGenAISettings.create(
```

Comment:
Should the `ai_model_id` be added to the `OnnxGenAISettings` -- one can pass it in via the constructor, but we should also try to read it from the env vars/.env file?